### PR TITLE
Add usage docs

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Obvious: make our project usable.

## Proposal

Revamp the whole documentation to a short README and a more complete documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,215 +1,125 @@
...
+with the collection of FUN services up and running. You can change the customer
+/ environment to deploy by overriding default environment variables:
 
-In development with MiniShift you can use the helper script:
+```bash
+$ bin/init -e "env_type=staging customer=campus"
+```
 
-    $ bin/ansible-playbook create_config.yml -e "env_type=development"
+If you want to run a new deployment for this project, there is also an helper
```
Comment (sampaccoud) [inline README.md]: s/an/a

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Arnold
+
+Arnold's is the place where all FUN's applications are assembled and deployed to
```
Comment (sampaccoud) [inline docs/index.md]: I would say remove the 's to Arnold

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+1.  [Requirements](./requirements.md)
+2.  [Install MiniShift](./minishift.md)
+3.  [Install & configure Arnold](./install.md)
+
+### II. Developer guide
+
+1.  [Concepts overview](./concepts.md)
+2.  [Getting started](./getting_started.md)
+3.  [Using Ansible playbooks](./playbooks.md)
+4.  [Working with routes](./routes.md)
```
Comment (sampaccoud) [inline docs/index.md]: working with routes -> 404

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+definition templates (Jinja2). We take advantage of the `openshift_raw` Ansible
+module to make Ansible talk with OpenShift.
+
+As a DevOps using this project, you will need to adapt OpenShift object
+templates to suite your needs or constraints and run playbooks to push your
+changes to your OpenShift instance that orchestrate your OpenEdx services.
+
+## Documentation
+
+### I. Installation
```
Comment (sampaccoud) [inline docs/index.md:24]: sort the doc files in a folder structure that follows this index?

File: docs/requirements.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Check that your `oc` client is ready to talk to your OpenShift console _via_:
+
+```bash
+# login to OpenShift console
+$ oc login https://openshift.mydomain.com:8443 --username=foo --password=xxx
+```
+
+with:
+
+* `openshift.mydomain.com`: the domaine name pointing to your running OpenShift
```
Comment (sampaccoud) [inline docs/requirements.md]: s/domaine/domain

File: docs/playbooks.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_project.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_secrets.yml`
+
+The `create_secrets.yml` playbook creates OpenShift secrets that may store
+sensible credentials. It uses vaulted Yaml definition files stored in
```
Comment (sampaccoud) [inline docs/playbooks.md]: s/sensible/sensitive (faux ami in french)

File: docs/playbooks.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_objects.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_acme.yml`
+
+We use this playbook to automatically generate [Let's
+Encrypt](https://letsencrypt.org) SSL certificates for every deployment (if the
+current `env_type` is not `development`).
```
Comment (sampaccoud) [inline docs/playbooks.md]: The certificates issued are automatically renewed by the service running deployed to OpenShift.

File: docs/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+functional on your system, it's time to play!
+
+> **Disclaimer**: the following tutorial has been written using MiniShift in
+> development environment. We will update this documentation soon for production
+> environments.
+
+## Using development environment
+
+Before running our first playbook, we need to start MiniShift and login to
+MiniShift's console _via_ the `oc login` command. This can be achieve with a
```
Comment (sampaccoud) [inline docs/getting_started.md]: s/achieve/achieved

File: docs/concepts.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+* `feature_title`: the slugified title of the feature. Only applicable when the
+  `environment` is equal to `feature`.
+
+The application name is used as a subdomain of the `domain_name` defined in
+`group_vars/all/openshift_routes.yml` (_e.g._ `example.com`) to compose urls,
+except for production environments where urls are set to the customer's domain.
+
+Here are some examples of valid urls on the `dev.openfun.fr` platform:
+
+* staging-campus-cms.dev.openfun.fr
```
Comment (sampaccoud) [inline docs/concepts.md]: replace dev.openfun.fr by "example.com" ?

File: docs/index.md (hunk 3 above)
Comment (jmaupetit) [reply docs/index.md]: Done

File: docs/index.md (hunk 4 above)
Comment (jmaupetit) [reply docs/index.md:24]: Done

**Comment by jmaupetit** [comment]
Hey @sampaccoud I've addressed all your comments. Tell me when you think we can ship it!

**Comment by sampaccoud** [comment]
Now  !

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**